dev/tests/api-functional/testsuite/Magento/Catalog/Api/CategoryRepositoryTest.php

```diff
@@ -476,5 +476,8 @@ public function testSaveDesign(): void
         }
         //We don't have permissions to do that.
         $this->assertEquals('Not allowed to edit the category\'s design attributes', $exceptionMessage);
+
+        // delete category to clean up auto-generated url rewrites
+        $this->deleteCategory($result['id']);
```

Comment:
Seems like this will not work in case if there will be some exception before.
I would suggest create some private property where you’ll store all data that need to be cleaned and do cleanup in the tearDown method.
Also please do the same in other tests